Src/GoogleApis.Auth.DotNet4.Tests/OAuth2/DefaultCredentialProvider/DefaultCredentialProviderTests.cs

```diff
@@ -0,0 +1,217 @@
+
+        #region UserCredential
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_UserCredentials_FromEnvironmentVariable()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, DummyUserCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (!(credential is UserCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type UserCredential. Obtained {0} instead.", credential.GetType()));
+            }
+        }
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_UserCredentials_FromWellKnownFileLocation()
+        {
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetFileContents(WellKnownCredentialFilePath, DummyUserCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (!(credential is UserCredential))
+            {
+                Assert.Fail(string.Format("Expected credential of type UserCredential. Obtained {0} instead.", credential.GetType()));
+            }
+        }
+
+        #endregion
+
+        #region ServiceAccountCredential
+
+        [Test]
+        public void TestGetApplicationDefaultCredentials_ServiceAccountCredentials_FromEnvironmentVariable()
+        {
+            string credentialFilepath = "TempFilePath.json";
+            TestDefaultCredentialProvider credentialProvider = new TestDefaultCredentialProvider();
+            credentialProvider.SetEnvironmentVariable(CredentialEnvironmentVariable, credentialFilepath);
+            credentialProvider.SetFileContents(credentialFilepath, DummyServiceAccountCredentialFileContents);
+
+            ICredential credential = credentialProvider.GetApplicationDefaultCredential();
+
+            if (credential is IScopableCredential)
+            {
+                IScopableCredential scopableCredential = credential as IScopableCredential;
```

Comment:
by the way, you should do the following:
var scopableCredential = credential as IScopableCredential;
if (scopableCredential == null)
{
}
else
{
}

Then you don't need to check "is" and then add add "as"